File: CPP/6 Kyu/remove_the_parentheses.cpp
#include <string>
using namespace std;
string remove_parentheses(const string &str)
{
    string  copy = "";
  
    int cont = 0;
    
    for(auto x:str){
      if (x!= '(' && cont==0){
        copy+=x;
        
      }else if(x=='('){
        cont+=1;
        
      }else if(x == ')'){
        cont-=1;
      }
    }
    return copy;
}

File: Rust/6 Kyu/remove_the_parentheses.rs
fn remove_parentheses(s: &str) -> String {
    
    let mut copy =String::from("");
    
    let mut cont = 0;
    let b = s;
    for x in b.chars(){
        if x != '(' && cont ==0{
            copy = copy.to_owned()+ &String::from(x);
        }else if x == '('{
            cont+=1;
            
        }else if x==')'{
            cont-=1;
        }
    }
    copy
    
}